torch/distributed/elastic/agent/server/api.py

```diff
@@ -557,12 +557,20 @@ def _rendezvous(self, worker_group: WorkerGroup) -> None:
         worker_group.group_world_size = group_world_size
 
         if group_rank == 0:
-            self._set_master_addr_port(
-                store,
-                spec.master_addr,
-                spec.master_port,
-                spec.local_addr,
-            )
+            # pyre-ignore [16]: Undefined attribute: torch._C._distributed_c10d.Store` has no attribute `check
+            if (
+                os.getenv("TORCH_SHARE_RDZV_TCP_STORE", "0") == "1"
+                and spec.rdzv_handler.static
+                and store.check(["MASTER_ADDR", "MASTER_PORT"])
```

Comment:
I'm not a fan of using the store to pass data between the rendezvous and agent. Since we're adding new methods/properties to the rendezvous class can we add one to get the master addr + port instead? That could also help clean up the existing static rendezvous logic (i.e. nuke spec.master_addr/master_port)